generate_sliding_window_chart.py

```
@@ -0,0 +1,87 @@
+
+        plt.plot(range(start_seconds, end_seconds + 1), rolling_average[start:end].values, color='red', linewidth=2)
+
+        plt.annotate(f"Start: {start_time}\nEnd: {end_time}",
+                     (midpoint, rolling_average.max() * 0.8),
+                     textcoords="offset points", xytext=(0, 10), ha='center', color='red',
+                     bbox=dict(facecolor='white', edgecolor='red', boxstyle='round,pad=0.5'))
+
+    plt.savefig(output_image)
+    plt.close()
+
+def generate_sliding_window_charts(input_csv, window_sizes=[10, 30, 60], increment_threshold=2, csv_folder='csv_files', chart_folder='charts'):
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)